### DISCUSSION TITLE
Announce: Go parser

### DISCUSSION THREAD

**Discussion by alecthomas** [kdl-org/kdl#97] [General]
Hi there,

I've written a [Go parser](https://github.com/alecthomas/kdl) for KDL. It's able to parse all of the [examples](https://github.com/kdl-org/kdl/tree/main/examples).

It currently doesn't do any marshalling to/from Go structs, nor is it able to serialise a Document to text, but it seems to parse okay. Is there a more comprehensive test suite available?

All the best,
Alec

**Comment by zkat** [comment]
There's not much of a test suite right now, unfortunately. Hopefully I'll be able to build one as part of the new kdl-rs rewrite I'm working on.

Would you like to make a PR to this repo and to the site to list your Go parser? :)

**Reply by alecthomas** [reply]
I did subsequently come across [this](https://reposhub.com/rust/utilities/zkat-kdl.html). While I'm not sure where it originates, it serves quite well as a mini test suite as it covers most of the KDL constructs! It picked up a few bugs in my implementation, which I'll hopefully fix today.

> Would you like to make a PR to this repo and to the site to list your Go parser? :)

Will do, thanks :)

**Comment by larsgw** [comment]
There's a Java test suite but you might need to serialise output to validate https://github.com/hkolbeck/kdl4j/tree/trunk/src/test/resources

**Reply by alecthomas** [reply]
Oh that's perfect, thanks @larsgw .
